src/cryptoadvance/specter/tor_daemon.py

```diff
@@ -84,15 +95,15 @@ def is_port_open(self):
             return False
 
     def stop_tor_daemon(self):
-        timeout = 50  # in secs
+        # It is possible for the Tor process to terminate unexpectedly without updating self.tor_daemon_proc
+        if not self.is_running():
+            return
         if self.tor_daemon_proc:
+            # This double approach ensures that the Tor child process is terminated on all levels (within Specter and on the OS level)
```

Comment:
I don't understand those two levels. What means "within Specter"? And what's wrong with the below code?

Reply:
Yeah, granted, could very well just be on MacOS. I only tested this on MacOS. I will include that in the comment.

Reply:
Basically, this is "within Specter": `self.tor_daemon_proc.terminate()` and this on the OS level: `subprocess.run(["pkill", "-f", cmdline_args])`